clinica/iotools/utils/pipeline_handling.py

```diff
@@ -58,10 +60,114 @@ def t1_freesurfer_pipeline(caps_dir, df, freesurfer_atlas_selection=None, **kwar
     Returns:
          final_df: a DataFrame containing the information of the bids and the pipeline
     """
+
     from clinica.iotools.converters.adni_to_bids.adni_utils import (
         replace_sequence_chars,
     )
     from clinica.utils.stream import cprint
+    from pathlib import Path
+
+    # Ensures that df is correctly indexed
+    if "participant_id" in df.columns.values:
+        df.set_index(["participant_id", "session_id"], inplace=True, drop=True)
+
+    subjects_dir = Path(caps_dir) / "subjects"
+
+    pipeline_df = pd.DataFrame()
+
+    for participant_id, session_id in df.index.values:
+
+        ses_path = subjects_dir / participant_id / session_id
+        mod_path = ses_path / "t1"
+
+        try:
+            flag_skip = False
+            long_id = next(mod_path.glob("long*")).name
+        except StopIteration:
+            cprint(
+                f"Could not find a longitudinal dataset for participant {participant_id} {session_id}",
+                lvl="warning",
+            )
+            flag_skip = True
+
+        if flag_skip:
+            continue
+
+        mod_path = mod_path / long_id / "freesurfer_longitudinal" / "regional_measures"
+
+        ses_df = pd.DataFrame(
+            [[participant_id, session_id]], columns=["participant_id", "session_id"]
+        )
+        ses_df.set_index(["participant_id", "session_id"], inplace=True, drop=True)
+
+        if mod_path.exists():
+            # Looking for atlases
+            atlas_paths = sorted(
+                (mod_path).glob(f"{participant_id}_{session_id}_*volume.tsv")
+            )
+            for atlas_path in atlas_paths:
+                if "-wm_volume" not in str(atlas_path) and "-ba_volume" not in str(
+                    atlas_path
+                ):
```

Comment:
```suggestion
                if not any(
                    s in atlas_path.name 
                    for s in ["-wm_volume", "-ba_volume"]
                ):
```

Reply:
maybe more elegant but not more readable

Reply:
no?

Reply:
Your call. Tomorrow if you need to add another case (say "-foo_volume"), the functional form with `any` is simpler to update (just add the new case to the list) than the imperative form. You can also use DeMorgan's theorem to change the `any` form to an `all` form if it turns out to be clearer to read (it might be).

Reply:
All right will keep it for now. Noted for cases with more options.

Reply:
It's also more explicit to test `atlas_path.name` (or even `atlas_path.steam` maybe?) so that we know where the expected pattern is in the full path.

```suggestion
                if "-wm_volume" not in atlas_path.name and "-ba_volume" not in atlas_path.name:
```

Reply:
Added stem.